app/scripts/lib/util.ts

```diff
@@ -178,8 +164,8 @@ export const generateRandomId = () => {
   return result;
 };
 
-export const isValidDate = (d) => {
-  return d instanceof Date && !isNaN(d);
```

Comment:
I don't think the && !isNaN(d) was ever possible to hit because anything that is an instance of Date will not be isNaN, even if constructing Date with a negative number